### PR TITLE
Applications: nrf5340_audio: Doc and sample.yaml update

### PR DESCRIPTION
**Pull request by koffes**
OCT-3067
Added more explicitly the use of flags in the doc. Cleaned up sample.yaml

### CODE REVIEW

File: applications/nrf5340_audio/doc/configuration.rst [hunk 1]
```diff
@@ -9,12 +9,28 @@ Configuring the nRF5340 Audio applications
...
 
+By default, if you have not made any changes to :file:`.conf` files at :file:`applications/nrf5340_audio/`, the nRF5340 build script tries to build the CIS/unicast application in the CIS unidirectional mode as a headset. ( ``CONFIG_TRANSPORT_CIS=y`` and ``CONFIG_AUDIO_DEV=2``).
+
+Selecting gateway or headset build
+**********************************
+
+The nRF5340 Audio applications can be built for either the gateway or the headset role.
+
+The gateway is a common term used for the base device.
+An Auracaster/broadast source or unicast client is a gateway.
```
Comment (rick1082) [inline applications/nrf5340_audio/doc/configuration.rst:20]: Aurcast/broadcast

File: applications/nrf5340_audio/doc/configuration.rst [hunk 2]
```diff
@@ -26,6 +42,11 @@ The walkie-talkie demo uses one or two bidirectional streams from the gateway to
 The PDM microphone is used as input on both the gateway and headset device.
 To switch to using the walkie-talkie, set the ``CONFIG_WALKIE_TALKIE_DEMO`` Kconfig option to ``y``  in the :file:`applications/nrf5340_audio/prj.conf` file (for the debug version) or in the :file:`applications/nrf5340_audio/prj_release.conf` file (for the release version).
 
+Enabling the Auracast/Broadcast mode
+====================================
+
+If you want to work with Auracasters/broadcast sources and sinks, set the ``CONFIG_TRANSPORT_BIS`` Kconfig option to ``y`` in the :file:`applications/nrf5340_audio/prj.conf` file.
```
Comment (rick1082) [inline applications/nrf5340_audio/doc/configuration.rst:48]: Auracast or Auracast transmitter

**Review by rick1082 — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/doc/configuration.rst [hunk 3]
```diff
@@ -9,12 +9,28 @@ Configuring the nRF5340 Audio applications
 
 |config|
 
+By default, if you have not made any changes to :file:`.conf` files at :file:`applications/nrf5340_audio/`, the nRF5340 build script tries to build the CIS/unicast application in the CIS unidirectional mode as a headset. ( ``CONFIG_TRANSPORT_CIS=y`` and ``CONFIG_AUDIO_DEV=2``).
```
Comment (alexsven) [inline applications/nrf5340_audio/doc/configuration.rst:12]:
```suggestion
By default, if you have not made any changes to :file:`.conf` files at :file:`applications/nrf5340_audio/`, the nRF5340 build script will build the CIS/unicast application in the CIS unidirectional mode as a headset. ( ``CONFIG_TRANSPORT_CIS=y`` and ``CONFIG_AUDIO_DEV=2``).
```

File: applications/nrf5340_audio/doc/configuration.rst [hunk 4]
```diff
@@ -9,12 +9,28 @@ Configuring the nRF5340 Audio applications
...
+
+The gateway is a common term used for the base device.
+An Auracaster/broadast source or unicast client is a gateway.
+This device is often used with USB or analog jack input.
+Often, but not always, the gateway is the largest or most stationary device, and is commonly the Bluetooth central if applicable.
+Gateway is selected by adding ``CONFIG_AUDIO_DEV=1`` to the prj.conf file.
+
+Oppositely, a headset is a commmon term for headphones, headsets, speakers or earbuds, acting as unicast server or broadcast sink.
+With reference to the gateway, the headset is often the smallest and most portable device, and is commonly the Bluetooth peripheral if applicable.
+Headset build is identified with ``CONFIG_AUDIO_DEV=2`` and is the default config.
```
Comment (alexsven) [inline applications/nrf5340_audio/doc/configuration.rst:27]:
```suggestion
Gateway is selected by adding ``CONFIG_AUDIO_DEV=2`` to the prj.conf file.

Oppositely, a headset is a common term for headphones, headsets, speakers or earbuds, acting as unicast server or broadcast sink.
With reference to the gateway, the headset is often the smallest and most portable device, and is commonly the Bluetooth peripheral if applicable.
Headset build is identified with ``CONFIG_AUDIO_DEV=1`` and is the default config.
```

Comment (alexsven) [reply applications/nrf5340_audio/doc/configuration.rst:27]:
```
config AUDIO_DEV
	int "Select which device type to compile for. 1=HEADSET or 2=GATEWAY"
	range 1 2
	default 1
	help
	  Setting this variable to 1 selects that the project is compiled
	  as a HEADSET device.
	  Setting to 2 will compile as a GATEWAY.
```
This is from the Kconfig

**Review by alexsven — CHANGES_REQUESTED** [review]

**Review by alexsven — APPROVED** [review]

**Review by rick1082 — APPROVED** [review]

**Review by eivindj-nordic — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Comment by greg-fer** [comment]
@koffes , please note I pushed with force, so make sure to pull my changes if you need to make further edits.

**Pull request merged**